backend/dataall/cdkproxy/stacks/dataset.py

```diff
@@ -389,53 +309,94 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
                 iam.AccountPrincipal(os.environ.get('CURRENT_AWS_ACCOUNT')),
                 iam.AccountPrincipal(dataset.AwsAccountId),
                 iam.ArnPrincipal(
-                    f'arn:aws:iam::{dataset.AwsAccountId}:role/{pivot_role_name}'
+                    f'arn:aws:iam::{dataset.AwsAccountId}:role/{self.pivot_role_name}'
                 ),
             ),
         )
         dataset_admin_policy.attach_to_role(dataset_admin_role)
 
-        glue_db_handler_arn = ssm.StringParameter.from_string_parameter_name(
+        # Datalake location custom resource: registers the S3 location in LakeFormation
+        # Get the lambda arn from SSM, this Lambda is created as part of the environment stack
+        # It replaces the following (just because it causes Cfn issues when handling upgrades of pivotRole)
+
+        # storage_location = CfnResource(
```

Comment:
I think I have already mentioned it, what may happen during the migration since I have encountered it by myself.
Removing this AWS::LakeFormation::Resource resource and adding a Custom Resource within the same process of updating the dataset stack causes a race condition since:
1. deleting AWS::LakeFormation::Resource from the cdk code triggers deletion of the data location resource from the lake ormation
2. adding a custom resource or actually replacing AWS::LakeFormation::Resource with a custom resource creates or updates the data location resource

Then the output of an update of the dataset stack depends on the order in which 1 & 2 occurs, in the worst case which happened to me (2 executed at first, 1  as second step):
2. replacing AWS::LakeFormation::Resource with a custom resource updated the datalake location with a new iam role
1. although updated correctly, removal of AWS::LakeFormation::Resource triggers deletion of the datalake location
As a result of this update, the datalake location is not anymore registered in the lake formation.

It is due to both AWS::LakeFormation::Resource and Custom Resource pointing to the same DataLakeLocation which can be registered just once.

I just want to raise it as a risk for the migration from the manual to cdk managed approach.
To mitigate it as a first step of the migration we need to update all the dataset stacks setting the AWS::LakeFormation::Resource deletion policy to RETAIN. Then regardless of the order we get rid of the risk of deleting the datalake location.

Reply:
Weird that I didn't happen for your tests. It depends on the order so you may be more lucky than I am. 

I just changed the deletion policy to RETAIN (separate commit), deployed the changes to the AWS, then executed stacks-updater task using aws-cli. Then in the yet another next commit I proceeded with removal of the original DataLakeLocation from the cdk code and replacing it with the Custom Resource, but it wasn't very convenient as you may imagine :D

' I am also thinking, maybe we can check in the on_update custom resource to create if not created. That way if in the first update (on_create) it fails, in the second one it will succeed. ' - I was also thinking about it, then the update would check if the datalakelocation is still there, if not, would recreate it. That would be more convenient.

But to be sure that the state of the aws resources is as expected we would need to execute update stacks task two times during the migration from manual to automated approach for the pivot role. So we need to mention it in the migration steps.

Reply:
Worst case scenario the daily task would trigger the update, so it would be a mismatch of a couple of hours

Reply:
yeah just to mention about this risk